plugin/src/main/java/net/dzikoysk/funnyguilds/feature/notification/bossbar/provider/DefaultBossBarProvider.java

```diff
@@ -22,42 +21,36 @@ public DefaultBossBarProvider(User user) {
 
     @Override
     public void sendNotification(String text, BossBarOptions options, int timeout) {
-        Player player = user.getPlayer();
-
-        if (player == null) {
-            return;
-        }
-
-        this.bossBar.setTitle(text);
-        this.bossBar.setColor(options.getColor());
-        this.bossBar.setStyle(options.getStyle());
-        options.getFlags().forEach(this.bossBar::addFlag);
-        this.bossBar.setVisible(true);
-
-        if (!this.bossBar.getPlayers().contains(player)) {
-            this.bossBar.addPlayer(player);
-        }
-
-        if (this.hideBossBarTask != null) {
-            this.hideBossBarTask.cancel();
-        }
-
-        this.hideBossBarTask = Bukkit.getScheduler().runTaskLaterAsynchronously(FunnyGuilds.getInstance(), () -> {
-            this.bossBar.removePlayer(player);
-            this.bossBar.setVisible(false);
-        }, 20L * timeout);
+        user.getPlayer()
+                .peek(player -> {
```

Comment:
Taki peek to można w tej linijce wyżej dać żeby bez sensu nie robić schodków - szczególnie że już tak robiłeś wcześniej